CRM/Contribute/BAO/Premium.php

```diff
@@ -171,22 +171,11 @@ public static function getPremiumProductInfo() {
   /**
    * Convert key=val options into an array while keeping
    * compatibility for values only.
+   *
+   * @deprecated
    */
   public static function parseProductOptions($string) : array {
-    $options = [];
-    $temp = explode(',', $string);
-
-    foreach ($temp as $value) {
-      $parts = explode('=', $value, 2);
-      if (count($parts) == 2) {
-        $options[trim($parts[0])] = trim($parts[1]);
-      }
-      else {
-        $options[trim($value)] = trim($value);
-      }
-    }
-
-    return $options;
+    return CRM_Utils_CommaKV::unserialize($string);
```

Comment:
It may be hard to see on a skim, but this (REF) is just renaming -- it's merely moved to the `CRM/Utils/` folder. 

(*Keeping it in here would lead to a dependency-loop -- where `CRM_Core_DAO` would depend on `CRM_Contribute_BAO` which already depends on `CRM_Core_DAO`. That would be bad form...*)